internal/overlord/servstate/handlers.go

```diff
@@ -499,14 +499,18 @@ func getAction(config *plan.Service, success bool) (action plan.ServiceAction, o
 func (s *serviceData) sendSignal(signal string) error {
 	switch s.state {
 	case stateStarting, stateRunning:
+		sig := unix.SignalNum(signal)
+		if sig == 0 {
+			return fmt.Errorf("invalid signal name %q", signal)
+		}
 		logger.Noticef("Sending %s to service %q", signal, s.config.Name)
-		err := syscall.Kill(-s.cmd.Process.Pid, unix.SignalNum(signal))
+		err := syscall.Kill(-s.cmd.Process.Pid, sig)
```

Comment:
This is incorrect for your usage, this signals the process group (-pid is a process group), which I think only makes sense in service stop.

I think we need to review this anyway for service stop:
- Stop should first signal SIGTERM to the service process (not the group, the process should SIGTERM its child processes)
- Wait the grace time
- Always send SIGKILL to the process group (even if the process exits, we need to signal the whole process group to ensure all child processes are killed).

@niemeyer does this sound right to you?

Reply:
This is a good call -- I fairly new to this signal handling and hadn't considered that. What you propose sounds reasonable. I'll try to do a bit more reading on it, and probably add this to the agenda for tomorrow night's spec review meeting.

Reply:
We discussed this on video and have a solution -- will follow up in a separate PR as it's not related to the send-signal API.